cogs/hosts.py

```diff
@@ -129,78 +130,82 @@ async def delhost(self, ctx, user: discord.Option(discord.Member, description="d
+        user_secondary_groups = await self.get_user_roles(user_id)
+
+        if 43 in user_secondary_groups:
+            await ctx.edit(content=f"{warning} {user.name} already has the temp-host group on the forum! Checking discord...")
+            time.sleep(3)
+        else:
+            message = await ctx.edit(content=f"{loading} Adding {user.name} to temp-host group on the forums...")
+            time.sleep(2)
+            await self.add_user_hosts(ctx, message, user_id, user_secondary_groups, user)
         
-        user_forum_id = result[0]
 
-        #
-        # check if there's an event on the calendar for today
-        # 
-        print("has account linked")
-        now = datetime.datetime.now()
-        calendar_today = "{:02d}".format(now.year) + "{:02d}".format(now.month) + "{:02d}".format(now.day)
-        query = "SELECT thread_id FROM xf_andy_calendar WHERE calendar_date = :field_value"
-        values = {"field_value": calendar_today}
-        results = await database.fetch_all(query=query, values=values)
-
-        for result in results:
-            #check if the users forum account is linked to that thread in the calendar thread
-            try:
-                #select user_id from xf_thread where thread_id='49093'
-                query = "select user_id from xf_thread where thread_id = :field_value"
-                values = {"field_value": result[0]}
-                thread_creator_id = await database.fetch_one(query=query, values=values)
-
-            except:
-                print("Can't find the user who created the thread. This is a rare error only see in testing.")
-                await ctx.edit(content=f"{error} Can't find user who created the thread on the calendar. You'll need to be added manually. This is a rare error only see in testing.")
-                return
-            
-            #check if the thread found matches the user_id linked
-            if thread_creator_id[0] != user_forum_id:
-                print("User ID on thread does not match the discord ID linked on the forums.")
-                await ctx.edit(content=f"{error} You don't seem to have an event thread on the calendar or you aren't the one who created the thread!")
-                return
-
-            #add to role
-            user_secondary_groups = await self.get_user_roles(thread_creator_id[0])
-
-            if 43 in user_secondary_groups:
-                await ctx.edit(content=f"{warning} {ctx.author.name} already has the temp-hosts group on the forum!")
-                time.sleep(3)
-            else:
-                message = await ctx.edit(content=f"{loading} Adding {ctx.author.name} to temp-hosts group on the forums...")
-                time.sleep(2)
-                await self.add_user_temp_host(ctx, message, user_forum_id, user_secondary_groups, ctx.author)
-            break
+        message = await ctx.edit(content=f"{loading} Adding {user.name} to hosts group on discord...")
+        time.sleep(2)
```

Comment:
if you added this to not get rate limited; know that the discord library will automatically repsect rate limit headers from discord so theres no need to do this